### PR TITLE
[FEATURE] Spots Detail + Review connection

### LINKED ISSUE #8: [FU] Fetch spot detail (including reviews)

### PR DESCRIPTION
**Pull request by mar-cas3**
Completes firebase integration for fetching a spots details and its corresponding reviews. 

Closes #8 

Notes:
* There is a small delay between the actual navigation and the fetching of the spot. In the video you can see the it first shows all columns with 0%. It would be really cool to maybe animate the bars to go form 0 to their rating but that's probably complicated, we could look into maybe adding a loading icon. However, these usability enhancements can be applied once all backend integration is complete.
* Browse view currently always navigates to same detail while that backend integration is completed
* I think we could calculate the `spot.reviewData.stats` locally since we are already fetching all of the review data, so that would mean no need for an endpoint. However, if a spot with may reviews it could be an issue to do it on device, so LT the endpoint is a better solution
* Moved the `ratings` dictionary onto the `ViewModel` to simplify the view
* We still need to work on Firebase error handling, but I think for now its okay to ignore that hehe, lmk what you think and I can look into that. 

**Video**

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/61592394/66a0606f-165a-4441-ad69-6bb6f1fdffd3

### CODE REVIEW

File: FoodBookApp/UI/Views/Browse/BrowseView.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ struct BrowseView: View {
         ScrollView {
             Group {
                 ForEach(searchResults, id: \.self) { spot in
-                    NavigationLink(destination: SpotDetailView()){ // TODO: In the future this should have the SpotId as param
+                    NavigationLink(destination: SpotDetailView(spotId: "ms1hTTxzVkiJElZiYHAT")){ // TODO: @JuanDiego
```
Comment (mar-cas3) [inline FoodBookApp/UI/Views/Browse/BrowseView.swift:23]: This is what currently makes all details to go to MiCaserito. @juanyepesp once you have the backend integratoin this should probably be like `spot.id`.

File: FoodBookApp/UI/Views/SpotDetail/SpotDetailViewModel.swift [hunk 2]
```diff
@@ -12,56 +12,18 @@ import FirebaseFirestore //FIXME: delete later
...
+    var ratings: [String: Double] = ["Cleanliness": 0, "Waiting time": 0, "Service": 0, "Food quality": 0]
+    
+    private let repository: SpotRepository = SpotRepositoryImpl.shared
+    
+    init() {}
+    
+    @MainActor
+    func fetchSpot(spotId: String) async throws {
+        self.spot = try await repository.getSpotById(docId: spotId)
+        self.ratings = ["Cleanliness": self.spot.reviewData.stats.cleanliness / 5,"Waiting Time": self.spot.reviewData.stats.waitTime / 5,"Service": self.spot.reviewData.stats.service / 5,"Food quality": self.spot.reviewData.stats.foodQuality / 5]
```
Comment (mar-cas3) [inline FoodBookApp/UI/Views/SpotDetail/SpotDetailViewModel.swift:27]: Ratings moved to View Model. Divides by 5 so its in the proper scale to show in the View

**Review by Laurarestrepo03 — APPROVED** [review]
Perfect, Maria, thank you! I have a quick question regarding your PR comment. When you say we should calculate the stats locally, should that be done in the ViewModel?

**Comment by mar-cas3** [comment]
> Perfect, Maria, thank you! I have a quick question regarding your PR comment. When you say we should calculate the stats locally, should that be done in the ViewModel?

yes i think so, something similar to what is done with the ratings in that method. but i think we should discuss that with everyone. it might better for performance to have that in the backend, rn since we don’t have many reviews it will be fast but it that grows it could slow down the phone.

**Pull request merged**